src/main/java/ogc/rs/processes/tilesMetaDataOnboarding/TilesMetaDataOnboardingProcess.java

```diff
@@ -0,0 +1,465 @@
+                                    ? new BigInteger(contentLengthHeader)
+                                    : BigInteger.ZERO;
+                            if (fileSize.compareTo(BigInteger.ZERO) > 0) {
+                                LOGGER.debug("File exists and has size: {}", fileSize);
+                                promise.complete(true);
+                            } else {
+                                LOGGER.warn("File exists but is empty");
+                                promise.fail(S3_EMPTY_FILE_MESSAGE);
+                            }
+                            break;
+                        case 404:
+                            LOGGER.warn("File not found in S3");
+                            promise.fail(S3_FILE_EXISTENCE_FAIL_MESSAGE);
+                            break;
+                        default:
+                            LOGGER.error("Unexpected HTTP status code: {}", statusCode);
+                            promise.fail("Failed to check file existence. HTTP Status: " + statusCode);
+                            break;
+                    }
+                })
+                .onFailure(failed -> {
+                    LOGGER.error("Failed to get response from S3: {}", failed.getMessage());
+                    promise.fail(S3_FILE_EXISTENCE_FAIL_MESSAGE);
+                });
+
+        return promise.future();
+    }
+
+    /**
+     * Checks the existence of the collection and determines if it is a pure tile collection
+     * or a combination of feature and tile collection.
+     * <p>
+     * This method queries the database to check if the collection exists. If the collection exists, it further checks
+     * its type. If the collection does not exist, it marks the collection as a pure tile collection.
+     * </p>
+     *
+     * @param requestInput A JSON object containing collection details with a key {@code "resourceId"} for the collection identifier.
+     * @return A {@link Future<JsonObject>} containing the updated JSON object with the {@code "pureTile"} attribute
+     * indicating whether the collection is a pure tile collection (true) or not (false).
+     */
+    private Future<JsonObject> evaluateCollectionTileType(JsonObject requestInput) {
+        String collectionId = requestInput.getString("resourceId");
+        Promise<JsonObject> promise = Promise.promise();
+
+        // Check if the collection exists in the collection_details table
+        pgPool.preparedQuery(CHECK_COLLECTION_EXISTENCE_QUERY)
+                .execute(Tuple.of(collectionId))
+                .onSuccess(rowSet -> {
+                    // Check if the collection exists
+                    if (rowSet.rowCount() > 0 && rowSet.iterator().next().getBoolean("exists")) {
+                        // Collection exists, now check its type in the collection_type table
+                        pgPool.preparedQuery(GET_COLLECTION_TYPE_QUERY)
```

Comment:
If you use the above mentioned SQL query, you can get rid of variables `isFeature`, `isVector` and `isMap` and simplify the logic in a cleaner way.